Wasm of ocaml support (#10695)
ocaml/dune: doc/wasmoo.rst

.. _wasmoo:

***************************************
Wasm Compilation With Wasm_of_ocaml
***************************************

.. TODO(diataxis)

   This is an how-to guide.

Wasm_of_ocaml_ is a compiler from OCaml to WebAssembly (Wasm for
short). The compiler works by translating OCaml bytecode to Wasm code.

Compiling to Wasm is very similar to compiling to JavaScript. See
:doc:`jsoo` for more information.

Compiling to Wasm
=================

> Dune has full support building Wasm_of_ocaml libraries and executables transparently.

Reviewer: ```suggestion
Dune has full support building wasm_of_ocaml libraries and executables transparently.
```
I have the formatting of this noted as Wasm_of_ocaml at the beginning of sentences and in titles, and wasm_of_ocaml in the middle of sentences. If it should always be capitalised like this, let me know, and I'll update the style guide.

> There's no need to customize or enable anything to compile OCaml

Reviewer: ```suggestion
There's no need to customise or enable anything to compile OCaml
```

> Dune has full support building Wasm_of_ocaml libraries and executables transparently.
> There's no need to customize or enable anything to compile OCaml

Reviewer: ```suggestion
Dune has full support for building Wasm_of_ocaml libraries and executables transparently.
There's no need to customise or enable anything to compile OCaml
```
